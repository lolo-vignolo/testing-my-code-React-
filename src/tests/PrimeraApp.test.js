// import { render } from "@testing-library/react";
import React from "react";

import { shallow } from "enzyme"
import PrimerApp from "../PrimeraApp";

describe('IÂ´ll test my first App', () => {
    
    // test('this has to get a prop from index.js to render in my app', () => {
    //     const saludo = "Hola soy Lorenzo";
    //     const {getByText} = render (<PrimerApp saludo= {saludo} />)
    //     expect (getByText(saludo)).toBeInTheDocument();
    // })

    test('debe mostrar <PrimeraApp /> correctamente ', () => {

        const saludo = "Hola soy Lorenzo";

        const wrapper =shallow( <PrimerApp saludo={saludo} />)

        expect(wrapper).toMatchSnapshot();
        
    })
    

    test('debe mostrar un subtitulo', () => {
        
        const saludo = "Hola soy Lorenzo"; 
        const subtit =  "soy un subtitulo"
        const wrapper = shallow( <PrimerApp saludo={ saludo }
        subtitulo={subtit} />);
        const textoParrafo = wrapper.find("p");
        expect (textoParrafo).toBe(subtit);
        
    })
    


    
})
