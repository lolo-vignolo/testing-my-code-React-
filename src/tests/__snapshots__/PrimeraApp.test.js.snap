// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IÂ´ll test my first App debe mostrar <PrimeraApp /> correctamente  1`] = `
<Fragment>
  <h1>
    Hola soy Lorenzo
  </h1>
  <p>
    soy un subtitulo
  </p>
</Fragment>
`;
